xds/internal/balancer/clusterimpl/config.go

```diff
@@ -25,32 +25,33 @@ import (
 	"google.golang.org/grpc/serviceconfig"
 )
 
-type dropCategory struct {
+// DropConfig contains the category, and drop ratio.
+type DropConfig struct {
 	Category           string
 	RequestsPerMillion uint32
 }
 
-// lbConfig is the balancer config for weighted_target.
-type lbConfig struct {
-	serviceconfig.LoadBalancingConfig
+// LBConfig is the balancer config for cluster_impl balancer.
+type LBConfig struct {
+	serviceconfig.LoadBalancingConfig `json:"-"`
 
-	Cluster                    string
-	EDSServiceName             string
-	LRSLoadReportingServerName *string
-	MaxConcurrentRequests      *uint32
-	DropCategories             []dropCategory
-	ChildPolicy                *internalserviceconfig.BalancerConfig
+	Cluster                 string                                `json:"cluster,omitempty"`
+	EDSServiceName          string                                `json:"edsServiceName,omitempty"`
+	LoadReportingServerName *string                               `json:"lrsLoadReportingServerName,omitempty"`
```

Comment:
Is snake_case or camelCase preferred for JSON field names?
I know that we use snake_case in the JSON of the bootstrap config.

Reply:
Hmm, maybe they should all be snake_case.

Reply:
OK, so, it's tricky.

1. `json` and `jsonpb` do different things, due to the generated code
   - e.g.
     ```
     EdsServiceName       string   
     `
     protobuf:"bytes,5,opt,name=eds_service_name,json=edsServiceName,proto3" 
     json:"eds_service_name,omitempty"
     `
     ```
   - `json` would use `eds_service_name`, and `jsonpb` would use `edsServiceName`
1. service config uses camelCase, e.g.: https://github.com/grpc/proposal/blob/master/A2-service-configs-in-dns.md#canarying-changes

I'm going to keep camelCase here, to keep it similar to service config. Let me know if you disagree.

Reply:
Ack.